File: BotBuilderNodeJs/libraries/botframework-schema/src/sharepoint/actionButton.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { Action } from './action';

/**
 * Sharepoint action button
 */
export class ActionButton {
    private title: string;
    private action: Action;
    private id: string;
    private style: ActionButton.ActionStyle;
    
    /**
     * Initializes a new instance of the ActionButton class
     */
    public ActionButton() {
        // Do nothing
    }

    /**
     * Sets title property of type string
     */
    public set Title(title: string) {
        this.title = title;
    }

    /**
     * Gets title property of type string
     */
    public get Title(): string {
        return this.title;
    }

    /**
     * Sets parameters property of type Action
     */
    public set Action (action: Action) {
        this.action = action;
    }

    /**
     * Gets parameters property of type Action
     */
    public get Action(): Action {
        return this.action; 
    }

    /**
     * Sets id property of type string
     */
    public set Id (id: string) {
        this.id = id;
    }

    /**
     * Gets id property of type string
     */
    public get Id(): string {
        return this.id; 
    }

    /**
     * Sets style property of type string
     */
    public set Style (style: ActionButton.ActionStyle) {
        this.style = style;
    }

    /**
     * Gets style property of type string
     */
    public get Style(): ActionButton.ActionStyle {
        return this.style; 
    }
}

export namespace ActionButton
{
    export enum ActionStyle {
        Default = "default",
        Positive = "positive",
        Destructive = "destructive"
    }
}

File: BotBuilder DotNet/libraries/Microsoft.Bot.Schema/SharePoint/ActionButton.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
using System.Text;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using Newtonsoft.Json.Linq;

namespace Microsoft.Bot.Schema.SharePoint
{
    /// <summary>
    /// SharePoint action button.
    /// </summary>
    public class ActionButton
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="ActionButton"/> class.
        /// </summary>
        public ActionButton()
        {
            // Do nothing
        }

        /// <summary>
        /// This enum contains the different types of action styles available in the SPFx framework.
        /// </summary>
        public enum ActionStyle
        {
            /// <summary>
            /// Default
            /// </summary>
            [EnumMember(Value = "default")]
            Default,

            /// <summary>
            /// Positive
            /// </summary>
            [EnumMember(Value = "positive")]
            Positive,

            /// <summary>
            /// Destructive
            /// </summary>
            [EnumMember(Value = "destructive")]
            Destructive
        }

        /// <summary>
        /// Gets or Sets the title of type <see cref="string"/>.
        /// </summary>
        /// <value>This value is the title of the action button.</value>
        [JsonProperty(PropertyName = "title")]
        public string Title { get; set; }

        /// <summary>
        /// Gets or Sets the action of type <see cref="SharepointAction"/>.
        /// </summary>
        /// <value>This value is the action of the action button.</value>
        [JsonProperty(PropertyName = "action")]
        public SharepointAction Action { get; set; }

        /// <summary>
        /// Gets or Sets the id of type <see cref="string"/>.
        /// </summary>
        /// <value>This value is the id of the action button.</value>
        [JsonProperty(PropertyName = "id")]
        public string Id { get; set; }

        /// <summary>
        /// Gets or Sets the style of type <see cref="ActionStyle"/>.
        /// </summary>
        /// <value>This value is the style of the action button.</value>
        [JsonProperty(PropertyName = "style")]
        public ActionStyle Style { get; set; }
    }
}
